main.go

```diff
@@ -139,6 +139,8 @@ func setupControllers(mgr ctrl.Manager, certsReady chan struct{}) {
 		setupLog.Error(err, "unable to create webhook", "webhook", "JobSet")
 		os.Exit(1)
 	}
+
+	mgr.GetWebhookServer().Register("/mutate-v1-pod", &webhook.Admission{Handler: &jobset.PodAnnotator{Client: mgr.GetClient()}})
```

Comment:
is "/mutate-v1-pod" just the name? where do we say that this is for pods?

Reply:
ah I was following https://book.kubebuilder.io/reference/webhook-for-core-types but went ahead and changed this to match how we register the validating/defaulting webhook, and specifically made it `For(&corev1.Pod{})`